crates/node-builder/src/builder.rs

```diff
@@ -585,6 +589,65 @@ where
         debug!(target: "reth::cli", "calling on_component_initialized hook");
         on_component_initialized.on_event(node_components.clone())?;
 
+        // spawn exexs
+        let mut exex_handles = Vec::with_capacity(self.state.exexs.len());
+        for (id, exex) in self.state.exexs {
+            // create a new exex handle
+            let (handle, events, notifications) = ExExHandle::new(id.clone());
+            exex_handles.push(handle);
+
+            // create the launch context for the exex
+            let context = ExExContext {
+                head,
+                provider: blockchain_db.clone(),
+                task_executor: executor.clone(),
+                data_dir: data_dir.clone(),
+                config: config.clone(),
+                reth_config: reth_config.clone(),
+                events,
+                notifications,
+            };
+
+            // init the exex
+            let exex = exex.launch(context).await.unwrap();
```

Comment:
can we move this below span initialization, so that tracing logs from `new` are captured under the correct span?